### ISSUE TITLE
rootless container

### ISSUE THREAD

**Issue by H777K** [netbirdio/dashboard#406]
**Is your feature request related to a problem? Please describe.**
In some restricted environments its not allowed to run containers with root privileges. Currently i have to build the docker image by myself, which causes a higher maintaining effort.

**Describe the solution you'd like**
It would be nice to have the possibilty to run the container with a non root user (e.g. nobody (65534)). Maybe the image can be published also as unprivileged. For example `nginx` also has an `nginx-unprivileged` image. It would be also fine to have a config, which allows to run the container in an unprivileged mode.

**Describe alternatives you've considered**
IMO there are no alternatives: either the possibility is given to run the container as unprivilged or I have to customize and maintain it by myself.

**Comment by saule1508** [comment]
Hi,
We are using podman (on redhat it is the default) which is rootless and the image works just fine. We are using quadlet. The user running the container is netbird. In the home of netbird, in .config/container/systemd/dasboard.conf you can set something like this

```
[Unit]
Description=Podman Dashboard Container

[Container]
Image=<image from netbird >
Network=pasta
Environment=NETBIRD_MGMT_API_ENDPOINT=...
Environment=NETBIRD_MGMT_GRPC_API_ENDPOINT=...
Environment=AUTH_AUDIENCE=...
Environment=AUTH_CLIENT_SECRET=
Environment=AUTH_AUTHORITY=...
.. all Env
LogDriver=journald
PublishPort=...

[Service]
Restart=always

[Install]
# Start by default on boot
WantedBy=default.target
```
the systemctl --user daemon-reload and systemctl --user start dashboard
U need to enable lingering for the netbird user in this case (https://github.com/containers/podman/blob/main/troubleshooting.md)

**Comment by AWildBeard** [comment]
In more hardened environments on RHEL, this container image does not work. For example, using subuid's and subgid's forces the "root" user in the container to not have the ability to bypass filesystem access controls. This creates an error where when nginx starts in the container (as root), it can't read/write to `/var/lib/nginx`, `/var/log/nginx`, etc because those directories are owned by a `nginx` user in the container. This should be considered a bug. Hardened environments that prohibit user switching inside of containers, bypassing filesystem ACL's in containers, etc. are common. 

So in short, the container image relies on root's ability to bypass filesystem access control lists to run which should be considered a dangerous pattern to rely upon in containerized contexts where many external factors may prohibit such activity.

A potential fix on the users side, without changing the container image, is to override *each* directory that has ownership issues with tmpfs mounts owned by "root".

**Comment by lixmal** [comment]
Have you tried the rootless image? Or is this about the management server?
 https://github.com/netbirdio/netbird/pkgs/container/netbird/809259429?tag=rootless-latest

**Comment by doge-gif** [comment]
@lixmal I think this is about the mgmt server (since we're in its repo), and there is a prototype pr by @reimarstier in #411. Though the last activity was almost a year ago (maintainers overlooked)?
